### PR TITLE
Fixes:[T315134] implement search-bar in queue page

### PR DESCRIPTION
**Pull request by okerekechinweotito**
Fixes: [T315134](https://phabricator.wikimedia.org/T315134)

## Proposed Changes
- Implement search functionality for the queue page
- Enable search by Job ID, Book Title , Username , and Status
- Search is conducted onChange i.e immediately as you type

## Files Updated

- components/QueueTable.js  - lifted the allJobs fetch-call out from QueueTable to its parent pages/queue to better handle Search
- pages/queue.js - add input field for search , make the allJobs fetch , write an onSearch handler function and pass tableData props to QueueTable
- components/Footer.js - Adjust Footer Style to compensate for displacement caused by SearchBar

## Checklist 
- [x] Coding Conventions are followed.
- [x] Comments are used for Documenting the Code.
- [x] Correct File Names are mentioned.

## Current Look

-Search By Book Title
Screenshot (19)

-Search By Status - Completed
Screenshot (20)

-Search By Status - Failed
Screenshot (21)

-Search By Job ID
Screenshot (26)

-Inactive Look
Screenshot (25)

### CODE REVIEW

**Comment by okerekechinweotito** [comment]
@Parthiv-M @coderwassananmol 
Please review

File: pages/queue.js [hunk 1]
```diff
@@ -26,7 +62,36 @@ const Queue = ({ data }) => {
...
+              <div className="cdx-search-input cdx-search-input--has-end-button">
+                <div
+                  className="cdx-text-input cdx-text-input--has-start-icon"
+                  style={{ width: "100%" }}
+                >
+                  <input
+                    onChange={(e) => onSearch(e)}
+                    className="cdx-text-input__input"
+                    type="search"
+                    placeholder="Search By Job ID , Book Title , Username , or Status"
```
Comment (coderwassananmol) [inline pages/queue.js:83]: This should be "Search by Job ID, Title, Username or Status"

**Review by coderwassananmol — CHANGES_REQUESTED** [review]
**Observation:**
Let's say the queue table has multiple pages and I am not on the first page. I search for a title and it fetches me 3 results. However, I will see 0 results because the 3 results are on the first page. I have to manually go back to the first page to see the results. If this can be solved, we can merge this PR. :)

**Comment by okerekechinweotito** [comment]
@coderwassananmol 
I have fixed the bug: You can now search from any page and see results without having to manually go back to the first page

File: pages/queue.js (hunk 1 above)
Comment (okerekechinweotito) [reply pages/queue.js]: fixed

**Review by coderwassananmol — APPROVED** [review]
LGTM

**Pull request merged**